File: fizzbuzz_1.1.rb
class FizzBuzzOnePointOne
  attr_accessor :result, :limit
  private :result, :limit

  def initialize(limit)
    @limit = limit
    @result = ""
  end

  def display_series
    (1..limit).each { |x| define_number(x) }
    puts result
    result
  end

  private

  def define_number(x)
    if x % 3 == 0 && x % 5 == 0
      @result << "fizzbuzz "
    elsif x % 3 == 0
      @result << "fizz "
    elsif x % 5 == 0
      @result << "buzz "
    else
      @result << "#{x} "
    end
  end
end

FizzBuzzOnePointOne.new(100).display_series

File: fizzbuzz_1.1.js
var FizzBuzzOnePointOne = (function() {
  function FizzBuzzOne(limit) {
    this.limit = limit;
    this.result = "";
  };

  FizzBuzzOnePointOne.prototype.displaySeries = function() {
    for (var i = 1; i <= this.limit; i++) {
      this.defineOutput(i);
    }
  };

  FizzBuzzOnePointOne.prototype.defineOutput = function(x) {
    if (x % 3 === 0 && x % 5 === 0) {
      this.result += "fizzbuzz ";
    }
    else if (x % 3 === 0) {
      this.result += "fizz ";
    }
    else if (x % 5 === 0) {
      this.result += "buzz ";
    }
    else {
      this.result += x + " ";
    }
  };

  return FizzBuzzOne;
})();

fizzbuzz = new FizzBuzzOnePointOne(10000000);
fizzbuzz.displaySeries();
